ssl/statem/statem_srvr.c

```diff
@@ -4331,132 +4457,17 @@ CON_FUNC_RETURN tls_construct_new_session_ticket(SSL_CONNECTION *s, WPACKET *pkt
-        /*
-         * check if its the right response in the case it is a successful response
-         * as not every time the issuer certificate is available the check just
-         * uses the issuer name and the serial number from the current certificate
-         */
-        if (OCSP_response_status(resp) == OCSP_RESPONSE_STATUS_SUCCESSFUL) {
-            /*
-             * set a mark for the error queue her to be able to ignore errors
-             * happening because of test cases
-             */
-            ERR_set_mark();
-            if (((bs = OCSP_response_get1_basic(resp)) != NULL)
-                && ((sr = OCSP_resp_get0(bs, 0)) != NULL)) {
-                /* use the first single response to get the algorithm used */
-                cid = (OCSP_CERTID *)OCSP_SINGLERESP_get0_id(sr);
-
-                OCSP_id_get0_info(&respIssuerNameHash, &cert_id_md_oid, NULL, &respSerial, cid);
-                if (cert_id_md_oid != NULL)
-                    cert_id_md = EVP_get_digestbyobj(cert_id_md_oid);
-                else
-                    cert_id_md = EVP_sha1();
-
-                /* get serial number and issuer name hash of the certificate from the chain */
-                certSerial = X509_get0_serialNumber(x);
-                certIssuerName = X509_get_issuer_name(x);
-                certIssuerNameHash = ASN1_OCTET_STRING_new();
-                if (!X509_NAME_digest(certIssuerName, cert_id_md, md, &len) ||
-                    !(ASN1_OCTET_STRING_set(certIssuerNameHash, md, len))) {
-                    ASN1_OCTET_STRING_free(certIssuerNameHash);
-                    OCSP_BASICRESP_free(bs);
-                    ERR_clear_last_mark();
-                    return 0;
-                }
-
-                num = OCSP_resp_count(bs);
-                for (i = 0; i < num; i++) {
-                    sr = OCSP_resp_get0(bs, i);
-
-                    /* determine the md algorithm which was used to create cert id */
-                    sr_cert_id = (OCSP_CERTID *)OCSP_SINGLERESP_get0_id(sr);
-
-                    OCSP_id_get0_info(&respIssuerNameHash, NULL, NULL, &respSerial, sr_cert_id);
-
-                    if (!ASN1_INTEGER_cmp(certSerial, respSerial) &&
-                        !ASN1_OCTET_STRING_cmp(certIssuerNameHash, respIssuerNameHash))
-                        break;
-                }
-
-                ASN1_OCTET_STRING_free(certIssuerNameHash);
-                OCSP_BASICRESP_free(bs);
-
-                /*
-                 * if we did not find the right single response in the OCSP response we
-                 * construct an empty message
-                 */
-                if (i == num)
-                    resp = NULL;
-            }
-
-            /*
-             * in a test case a response without a basic response is used the error set
-             * could be ignored here
-             */
-            ERR_pop_to_mark();
-        }
-    }
-
     if (resp != NULL)
```

Comment:
I believe `resp == NULL` is now impossible and should probably fail with `ERR_R_INTERNAL_ERROR` or an assert or whatever is preferred these days.

If `resp` ever were `NULL`, I believe you all would output a syntax error.

Reply:
(This condition is now impossible.)